Gems/WhiteBox/Code/Source/SubComponentModes/EditorWhiteBoxTransformMode.cpp

```diff
@@ -6,50 +6,648 @@
+                    DrawFace(debugDisplay, whiteBox, polygonSelection->GetHandle(), ed_whiteBoxPolygonSelection);
+                    DrawOutline(debugDisplay, whiteBox, polygonSelection->GetHandle(), ed_whiteBoxOutlineSelection);
+                }
+            }
+            else if (auto edgeSelection = AZStd::get_if<EdgeIntersection>(&m_whiteBoxSelection->m_selection))
+            {
+                auto vertexHandles = Api::EdgeVertexHandles(*whiteBox, edgeSelection->GetHandle());
+                DrawPoints(debugDisplay, whiteBox, worldFromLocal, viewportInfo, vertexHandles, ed_whiteBoxVertexSelection);
+                if (m_edgeIntersection.value_or(EdgeIntersection{}).GetHandle() != edgeSelection->GetHandle())
+                {
+                    DrawEdge(debugDisplay, whiteBox, edgeSelection->GetHandle(), ed_whiteBoxOutlineSelection);
+                }
+            }
+            else if (auto vertexSelection = AZStd::get_if<VertexIntersection>(&m_whiteBoxSelection->m_selection))
+            {
+                if (m_vertexIntersection.value_or(VertexIntersection{}).GetHandle() != vertexSelection->GetHandle())
+                {
+                    auto handles = AZStd::array<Api::VertexHandle, 1>({ vertexSelection->GetHandle() });
+                    DrawPoints(debugDisplay, whiteBox, worldFromLocal, viewportInfo, handles, ed_whiteBoxVertexSelection);
+                }
+            }
+        }
+        debugDisplay.PopMatrix();
+        debugDisplay.DepthTestOff();
     }
 
     bool TransformMode::HandleMouseInteraction(
-        [[maybe_unused]] const AzToolsFramework::ViewportInteraction::MouseInteractionEvent& mouseInteraction,
+        const AzToolsFramework::ViewportInteraction::MouseInteractionEvent& mouseInteraction,
         [[maybe_unused]] const AZ::EntityComponentIdPair& entityComponentIdPair,
-        [[maybe_unused]] const AZStd::optional<EdgeIntersection>& edgeIntersection,
-        [[maybe_unused]] const AZStd::optional<PolygonIntersection>& polygonIntersection,
-        [[maybe_unused]] const AZStd::optional<VertexIntersection>& vertexIntersection)
+        const AZStd::optional<EdgeIntersection>& edgeIntersection,
+        const AZStd::optional<PolygonIntersection>& polygonIntersection,
+        const AZStd::optional<VertexIntersection>& vertexIntersection)
     {
+        WhiteBoxMesh* whiteBox = nullptr;
+        EditorWhiteBoxComponentRequestBus::EventResult(
+            whiteBox, m_entityComponentIdPair, &EditorWhiteBoxComponentRequests::GetWhiteBoxMesh);
+
+        const bool mouseOverMainpulator = ([&]() 
+            {
+                if(m_manipulator) 
+                {
+                    for(auto& manipulator: m_manipulator->GetManipulators()) 
+                    {
+                        if(manipulator->MouseOver()) 
+                        {
+                            return true;
+                        }
+                    }
+                }
+                return false;
```

Comment:
update: I see this is why the `GetManipulators` was added, I do think there's another way to solve this, let me have a think and get back to you...

Reply:
you can just add a MouseOver() method? not sure if that is the ideal solution, don't think there is a situation where you only want to see if a particular part of the manipulator is hovered over.

Reply:
Agreed, I think we can do something like this (discussed offline)

```
bool MouseOver()
{
    bool mouseOver = false;
    const auto mouseOverFn = [&mouseOver](BaseManipulator* manipulator)
    {
        mouseOver = manipulator->MouseOver() && mouseOver;
    };

    ProcessManipulators(mouseOverFn);

    return mouseOver;
}
```

Haven't tested this but I reckon something like this could work 🙂 (I hope! 😅)